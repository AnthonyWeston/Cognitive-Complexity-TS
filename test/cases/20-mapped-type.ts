type G<T extends string> = {
    [K in T]: true; // iteration
};

// iterative condition
type H<T extends string> = {
    [K in T]: K extends boolean ? "t" : "f";
};

// conditional iteration
type I<T extends string> = T extends boolean
    ? {
        [K in T]: boolean;
    }
    : never;
