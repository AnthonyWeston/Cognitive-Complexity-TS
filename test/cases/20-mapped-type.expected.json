{
    "20-mapped-type.ts": {
        "score": 6,
        "inner": [
            {
                "name": "G",
                "score": 1,
                "line": 1,
                "column": 1
            },
            {
                "name": "H",
                "score": 2,
                "line": 6,
                "column": 1
            },
            {
                "name": "I",
                "score": 3,
                "line": 11,
                "column": 1
            }
        ]
    }
}